### PR TITLE
groups: introduce enforce edge groups for orgs

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
Despite the feature flag inventory-groups-enabled is on, we can force some orgs to use edge groups via configuration.
- Fixed devicesview api doc return type
- Updated features to accept unleash context
- Added the feature flag "enforce-edge-groups"
- Added the function that check whenther to enfoce-edge-groups for an organization.
- Added enforce_edge_groups to devicesview return data, so that the UI have not to make new request when working with devicesview.
- Added a new endpoint  "/device-groups/enforce-edge-groups" that return whether edge groups usage is enforced or not.
- Covered all the new functionality code with unit-tests.

FIXES: https://issues.redhat.com/browse/THEEDGE-3552

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/routes/devicegroups.go [hunk 1]
```diff
@@ -890,3 +892,24 @@ func containsInt(s []uint, searchterm uint) bool {
...
+// @Accept       json
+// @Produce      json
+// @Success      200 {object} models.EnforceEdgeGroupsAPI
+// @Failure      400 {object} errors.BadRequest
+// @Router       /device-groups/enforce-edge-groups [get]
+func GetEnforceEdgeGroups(w http.ResponseWriter, r *http.Request) {
+	ctxServices := dependencies.ServicesFromContext(r.Context())
+	orgID := readOrgID(w, r, ctxServices.Log)
+	if orgID == "" {
+		// logs and response handled by readOrgID
```
Comment (mgold1234) [inline pkg/routes/devicegroups.go:909]: logs is missing?

File: pkg/routes/devices.go [hunk 2]
```diff
@@ -388,7 +396,7 @@ func GetDevicesView(w http.ResponseWriter, r *http.Request) {
 // @Param	 image_id           query int   	false "field: filter by image id"
 // @Param	 limit              query int    	false "field: return number of devices until limit is reached. Default is 100."
 // @Param	 offset             query int    	false "field: return number of devices beginning at the offset."
-// @Success      200  {object}  models.DeviceViewListAPI
+// @Success      200  {object}  models.DeviceViewListResponseAPI
```
Comment (mgold1234) [inline pkg/routes/devices.go:399]: why we need new end point here?

File: pkg/routes/devicegroups.go (hunk 1 above)
Comment (ldjebran) [reply pkg/routes/devicegroups.go:909]: we do need logs as it was handled by readOrgID as said in the comment

File: pkg/routes/devices.go (hunk 2 above)
Comment (ldjebran) [reply pkg/routes/devices.go:399]: there is no new end-point here , at this line I am fixing the result type .

**Review by mgold1234 — DISMISSED** [review]
/lgtm

**Comment by ldjebran** [comment]
put do not merge , to try other strategies

File: config/config.go [hunk 3]
```diff
@@ -371,6 +372,13 @@ func CreateEdgeAPIConfig() (*EdgeConfig, error) {
 		}
 	}
 
+	edgeConfig.EnforceEdgeGroupsOrgs = make([]string, 0)
+	for _, value := range strings.Split(options.GetString("ENFORCE_EDGE_GROUPS_ORGS"), " ") {
+		if value != "" {
+			edgeConfig.EnforceEdgeGroupsOrgs = append(edgeConfig.EnforceEdgeGroupsOrgs, value)
+		}
+	}
+
```
Comment (de1987) [inline config/config.go:381]: As we discussed, the feature flag could be handled by Unleash.
@ldjebran will investigate how to implement it by using a new strategy for the Unleash feature flag.

Comment (ldjebran) [reply config/config.go]: updated

**Comment by ldjebran** [comment]
@mgold1234 @de1987 addressed all comments , please revisit

**Review by mgold1234 — APPROVED** [review]
/lgtm

**Review by acosferreira — APPROVED** [review]
lgtm

**Pull request merged**